tests/publiccloud/transfer_repos.pm

```diff
@@ -33,10 +33,23 @@ sub run {
 
         $args->{my_instance}->retry_ssh_command(cmd => "which rsync || sudo zypper -n in rsync", timeout => 420, retry => 6, delay => 60);
 
+        # In Incidents there is INCIDENT_REPO instead of MAINT_TEST_REPO
+        # Those two variables contain list of repositories separated by comma
+        set_var('MAINT_TEST_REPO', get_var('INCIDENT_REPO')) unless get_var('MAINT_TEST_REPO');
+
+        my @repos = split(/,/, get_var('MAINT_TEST_REPO'));
+        s/http*:\/\/// for @repos;
+
+        # Create list of directories for rsync
+        my $directories;
+        $directories .= "$_\n" foreach @repos;
+        assert_script_run("echo -en '$directories' | tee /tmp/transfer_repos.txt");
+
         # Mitigate occasional CSP network problems (especially one CSP is prone to those issues!)
         # Delay of 2 minutes between the tries to give their network some time to recover after a failure
-        script_retry("rsync --timeout=$timeout -uvahP -e ssh ~/repos '$remote:/tmp/repos'", timeout => $timeout + 10, retry => 3, delay => 120);
-        $args->{my_instance}->ssh_assert_script_run("sudo find /tmp/repos/ -name *.repo -exec sed -i 's,http://,/tmp/repos/repos/,g' '{}' \\;");
+        # For rsync the ~/repos/./ means that the --relative will take efect after.
+        script_retry("rsync --timeout=$timeout -uvahPRd -e ssh --files-from /tmp/transfer_repos.txt ~/repos/./ '$remote:/tmp/repos/'", timeout => $timeout + 10, retry => 3, delay => 120);
```

Comment:
Since those are automated test runs I would omit the `-vP` flags. They spam the serial output with no real value IMHO. WDYT?

Reply:
I will get rid of `-v` but the `-P` is needed. OK for you?